api/src/test/resources/io/strimzi/api/kafka/model/KafkaConnector.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaConnector
+metadata:
+  name: test-source
+spec:
+  connectCluster:
```

Comment:
I wonder if we could have two options here:
1) to specify the connect cluster by URL
2) to specify the connect cluster as reference to the KafkaConnect resource

Reply:
2. seems the more idiomatic way when thinking of CRDs. Would there be any advantage of allowing 1.?

Reply:
@gunnarmorling it's slightly more flexible, for example if the connect cluster was not deployed by Strimzi or not running inside Kube.

Reply:
Yes, it's more flexible, is that needed, though? I.e. when would you deploy a connector as a CRD but then not KC itself, too? Essentially I'm wary of the consequences of using the URL. E.g. when changing the KC port later on, do all these references have to be adjusted, too?

Reply:
If you used the `url` property and later changed the port then yes, you'd have to adjust. Supporting both means the user can decide based on their needs. The documentation can explain the pros and cons.